File: selection_sort.py
def swap(x, y):
    temp = x
    x = y
    y = temp
    return x, y

def selection_sort(arr):
    size = len(arr)
    for i in range(size - 1):
        min_index = i
        for j in range(i + 1, size):
            if arr[j] < arr[min_index]:
                min_index = j
        arr[i], arr[min_index] = swap(arr[i], arr[min_index])

def main():
    arr = [64, 25, 12, 22, 11]

    print("Original array:", arr)

    selection_sort(arr)

    print("Sorted array:", arr)

if __name__ == "__main__":
    main()

File: selection_sort.c
#include <stdio.h>

void swap(int *x, int *y) {
    int temp = *x;
    *x = *y;
    *y = temp;
}

void selection_sort(int arr[], int size) {
    int i, j, min_index;
    for (i = 0; i < size - 1; i++) {
        min_index = i;
        for (j = i + 1; j < size; j++)
            if (arr[j] < arr[min_index])
                min_index = j;
        swap(&arr[min_index], &arr[i]);
    }
}

int main() {
    int arr[] = {64, 25, 12, 22, 11};
    int size = sizeof(arr) / sizeof(arr[0]);

    printf("Original array: ");
    for (int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    selection_sort(arr, size);

    printf("Sorted array: ");
    for (int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    return 0;
}
